File: src/ders8/rpgornekjava/Oyuncu.java
package ders8.rpgornekjava;

public class Oyuncu {

    String isim;
    int saglik = 100;

    public Oyuncu(String isim, int saglik) {
        this.isim = isim.trim().toUpperCase().charAt(0) + isim.trim().substring(1);
        this.saglik = saglik;
    }

    public Oyuncu(String isim) {
        this.isim = isim.trim().toUpperCase().charAt(0) + isim.trim().substring(1);
    }

    public void hasarAl(){
        System.out.println(isim + " hasar aldı!");
    }
}

File: src/ders8/rpgornekkotlin/Oyuncu.kt
package ders8.rpgornekkotlin

open class Oyuncu (
    var isim: String,
    var saglik: Int = 100
){

    init {
        isim = isim.first().uppercase() + isim.substring(1)
    }


    fun hasarAl(){
        println("${isim} hasar aldı!")
        println("Canı $saglik -> ${saglik - 10}")
        if (saglik <= 0){
            println("${isim.first().uppercase() + isim.substring(1)} öldü!")
        }
    }
}